### PR TITLE
Bugfix: Fixed reset button works with logs

### PR DESCRIPTION
**Pull request by aicioara**
**Motivation:** When processing offline logs, if I wait until the end of the log, pressing the reset button has no effect.

**Fix:** Rewinding the tape when reset is pressed

**Side effects in the API**: Added new `rewind()` point

**Style changes:**. `rewound()` is an adjective. Adjectives should have NO side effects. There should be no trailing spaces in the files (IDE automatically trimmed them and it's a bit hard to add them all back, if picky about that, I can make the effort)

**Behavior changes** `rewound()` no longer rewinds the tape. It is a pure function now. 

**Tests**: 
* Live - works
* Log - works
* Log with the `-r` option - works

**I was careful about:**
* All usages of `rewound()`

**Attentions:**
- I do not guarantee this does not break something else completely.

### CODE REVIEW

File: GUI/src/Tools/RawLogReader.cpp [hunk 1]
```diff
@@ -146,21 +146,25 @@ bool RawLogReader::hasMore()
     return currentFrame + 1 < numFrames;
 }
 
-bool RawLogReader::rewound()
-{
-    if(filePointers.size() == 0)
+
+void RawLogReader::rewind() {
+    if (filePointers.size() != 0)
```
Comment (mp3guy) [inline GUI/src/Tools/RawLogReader.cpp]: Please keep code style here (i.e. new line after braces)

File: GUI/src/Tools/RawLogReader.cpp [hunk 2]
```diff
@@ -146,21 +146,25 @@ bool RawLogReader::hasMore()
...
     {
-        fclose(fp);
-        fp = fopen(file.c_str(), "rb");
+        filePointers = {};
+    }
 
-        assert(fread(&numFrames, sizeof(int32_t), 1, fp));
+    std::cerr << "resetted\n";
```
Comment (mp3guy) [inline GUI/src/Tools/RawLogReader.cpp]: Don't introduce superfluous output to the terminal

File: GUI/src/MainController.cpp [hunk 3]
```diff
@@ -181,6 +181,7 @@ void MainController::launch()
                 delete eFusion;
             }
 
+            logReader->rewind();
```
Comment (mp3guy) [inline GUI/src/MainController.cpp]: I don't see any reason for this to get called here

Comment (aicioara) [inline GUI/src/MainController.cpp]: This is the bug fix. We rewind when we reset.

**Pull request merged**